packages/collector/test/tracing/messaging/nats-streaming/test.js

```diff
@@ -147,22 +147,22 @@ mochaSuiteFn('tracing/nats-streaming', function () {
             simple: false
           })
           .then(() => {
-            const receivedMessages = subscriberControls.getIpcMessages();
-            if (!withError) {
-              expect(receivedMessages).to.have.lengthOf.at.least(1);
-              expect(receivedMessages[receivedMessages.length - 1]).to.equal(uniqueId);
-            }
+            return retry(() => {
+              const receivedMessages = subscriberControls.getIpcMessages();
+              if (!withError) {
+                expect(receivedMessages).to.have.lengthOf.at.least(1);
+                expect(receivedMessages[receivedMessages.length - 1]).to.equal(uniqueId);
+              }
```

Comment:
this is actually the change that should make the test less flaky - receiving the message needs to happen inside the retry